### ISSUE TITLE
Receive error creating additive primitive on plane in 0.21.2, works fine in 0.20.1.

### ISSUE THREAD

**Issue by Druloric** [FreeCAD/FreeCAD#12713]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

><class 'SyntaxError'>: ('unterminated string literal (detected at line 3)', ('<string>', 3, 59, "App.getDocument('Unnamed').getObject('Box').Height='2.38 ''", 3, 59))

Creating a padded base (.75" x 9.25" x 28") after which adding an additive primitive (.75" x 9.25" x 28.5") to the surface receives this error.  The same attempt works fine in 0.20.1 release.

### Full version info

```shell
[code]
OS: Linux 6.7.7-arch1-1 (qtile)
Word size of FreeCAD: 64-bit
Version: 0.21.2.33771 (Git)
Build type: Release
Branch: makepkg
Hash: b9bfa5c5507506e4515816414cd27f4851d00489
Python 3.11.7, Qt 5.15.12, Coin 4.0.2, Vtk 9.3.0, OCC 7.7.2
Locale: English/United States (en_US)
[/code]
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
Could you try this in the weekly builds: https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds

**Comment by Roy-043** [comment]
If the unit system is Building US this may be related to #11345.

**Comment by maxwxyz** [comment]
@Druloric could you provide feedback?

**Comment by MiqCG** [comment]
Hi,

I have tested with FreeCAD 0.22 revision 38459: the pop-up window with the error message doesn't appear and the Additive Primitive Box can be created on the face of the padded sketch.
Due to lack of information I guessed that was a Primitive Box with US Customary units and no particular orientation of the parts, constraints and base face.

I attach the files created with 0.20.2, 0.21.1 (unsuccesful) and 0.22 rev 38459.
[Primitive.zip](https://github.com/user-attachments/files/16752061/Primitive.zip)


Image with the successful operation in 0.22
![Captura de pantalla_2024-08-26_20-00-04](https://github.com/user-attachments/assets/592fff08-f209-4e34-9500-012121349cdf)


Image of the error with FreeCAD 0.21.1:
![Captura de pantalla_2024-08-26_19-46-51](https://github.com/user-attachments/assets/41c387e2-476d-4402-b06e-d1cff63a1ab9)
